### ISSUE TITLE
Support IAM credentials

### ISSUE THREAD

**Issue by jhmartin** [elastic/curator#914]
Rather than require hard-coding of credentials, accept EC2 / ECS IAM role temporary credentials.  A sample of how this can be done is at https://github.com/sam-washington/requests-aws4auth/issues/14.

```
import boto3
from requests_aws4auth import AWS4Auth

session = boto3.Session()
credentials = session.get_credentials()
region = os.environ['AWS_REGION']

awsauth = AWS4Auth(credentials.access_key, credentials.secret_key, region, 'es', session_token=credentials.token)
```

**Comment by untergeek** [comment]
Unfortunately this is moot until AWS ES supports Curator natively.

**Comment by jhmartin** [comment]
https://github.com/Talend/curator/pull/1 appears to be sufficient to get the delete-indices action working, at least to the point it complains about auth for deletion:

```
2017-03-29 18:22:28,356 INFO      Deleting selected indices: [u'logstash-2017.03.28', u'logstash-2017.03.29']
2017-03-29 18:22:28,356 INFO      ---deleting index logstash-2017.03.28
2017-03-29 18:22:28,356 INFO      ---deleting index logstash-2017.03.29
2017-03-29 18:22:28,456 ERROR     Failed to complete action: delete_indices.  <class 'curator.exceptions.FailedExecution'>: Exception encountered.  Rerun with loglevel DEBUG and/or check Elasticsearch logs for more information. Exception: TransportError(403, u'{"Message":"User: anonymous is not authorized to perform: es:ESHttpDelete on resource: REDACTED"}')
```

**Comment by untergeek** [comment]
It will not be merged.  That has been stated on multiple occasions.  I'm not going to add a one-off to resolve that issue.

**Comment by untergeek** [comment]
The chief reason is that it would iterate over that call for every index, rather than as a single call.

**Comment by jhmartin** [comment]
Thank you for clarifying that.  I put in a request for AWS to natively support Curator, FWIW.

**Comment by untergeek** [comment]
This is firmly in the `adopt me` category.  I do not have an AWS ES cluster to test against.  This will require community contribution to be resolved.

**Comment by untergeek** [comment]
This should be addressed in #1084
